client/app/bundles/course/assessment/submission/pages/SubmissionEditIndex/index.jsx

```diff
@@ -82,6 +81,11 @@ class VisibleSubmissionEditIndex extends Component {
     dispatch(fetchSubmission(params.submissionId));
   }
 
+  componentWillUnmount() {
+    const { dispatch } = this.props;
+    dispatch(purgeSubmissionStore());
```

Comment:
Just curious, why do we need to `purgeSubmissionStore`?

Reply:
When leaving the page, the store still contains data from loading a submission page. If the user goes to another submission page, the redux store is not empty and the data may be mixed up. To be safe, we purge the store. Need a bit of refactoring to avoid purging so I am not keen to do it now.